subiquity/server/controllers/shutdown.py

```diff
@@ -23,42 +23,65 @@
 from subiquitycore.utils import arun_command, run_command
 
 from subiquity.common.apidef import API
+from subiquity.common.types import ShutdownMode
 from subiquity.server.controller import SubiquityController
 from subiquity.server.controllers.install import ApplicationState
 
 log = logging.getLogger("subiquity.controllers.restart")
 
 
-class RebootController(SubiquityController):
+class ShutdownController(SubiquityController):
 
-    endpoint = API.reboot
+    endpoint = API.shutdown
+    autoinstall_key = 'shutdown'
+    autoinstall_schema = {
+        'type': 'string',
+        'enum': ['reboot', 'poweroff']
+    }
 
     def __init__(self, app):
         super().__init__(app)
-        self.user_reboot_event = asyncio.Event()
-        self.rebooting_event = asyncio.Event()
+        self.user_shutdown_event = asyncio.Event()
```

Comment:
I think the event names are similar enough, and nuanced enough in their concept, to be worthy of comments explaining their intent.  Let me have a try
user_shutdown_event # the user has requested the shutdown
server_reboot_event # the server is ready for shutdown
shuttingdown_event # the actual shutdown is about to start